dot/parachain/availability-store/availabilitystore.go

```diff
@@ -57,33 +183,93 @@ func (av *AvailabilityStoreSubsystem) processMessages() {
 }
 
 func (av *AvailabilityStoreSubsystem) handleQueryAvailableData(msg QueryAvailableData) {
-	// TODO: handle query available data
+	result, err := av.availabilityStore.loadAvailableData(msg.CandidateHash)
+	if err != nil {
+		logger.Errorf("failed to load available data: %w", err)
+	}
+	msg.Sender <- result
 }
 
 func (av *AvailabilityStoreSubsystem) handleQueryDataAvailability(msg QueryDataAvailability) {
-	// TODO: handle query data availability
+	_, err := av.availabilityStore.loadMetaData(msg.CandidateHash)
+	if err != nil {
+		msg.Sender <- false
+	} else {
+		msg.Sender <- true
+	}
 }
 
 func (av *AvailabilityStoreSubsystem) handleQueryChunk(msg QueryChunk) {
-	// TODO: handle query chunk
+	result, err := av.availabilityStore.loadChunk(msg.CandidateHash, msg.ValidatorIndex)
+	if err != nil {
+		logger.Errorf("failed to load chunk: %w", err)
+	}
+	msg.Sender <- result
 }
 
 func (av *AvailabilityStoreSubsystem) handleQueryChunkSize(msg QueryChunkSize) {
-	// TODO: handle query chunk size
+	meta, err := av.availabilityStore.loadMetaData(msg.CandidateHash)
+	if err != nil {
+		logger.Errorf("load metadata: %w", err)
+	}
+	var validatorIndex uint32
+	for i, v := range meta.ChunksStored {
+		if v {
+			validatorIndex = uint32(i)
+			break
+		}
+	}
+
+	chunk, err := av.availabilityStore.loadChunk(msg.CandidateHash, validatorIndex)
+	if err != nil {
+		logger.Errorf("load chunk: %w", err)
+	}
+	msg.Sender <- uint32(len(chunk.Chunk))
 }
```

Comment:
No need to log errors here. Let's  return errors in all `handle` functions and log them in processMessages. 
That way we only log them once in the end.

will avoid adding this comment at other such instances.